tests/wpunit/AssetsTest.php

```diff
@@ -22,22 +22,53 @@ public function tearDown() {
 	/**
 	 * @test
 	 */
-	public function it_should_should_register_multiple_assets() {
+	public function it_should_register_multiple_assets() {
 		Asset::add( 'my-script', 'fake.js' )->register();
 		Asset::add( 'my-style', 'fake.css' )->register();
 
-		$this->assertTrue( Assets::init()->exists( 'my-script' ) );
-		$this->assertTrue( Assets::init()->exists( 'my-style' ) );
-		$this->assertTrue( wp_script_is( 'my-script', 'registered' ) );
-		$this->assertTrue( wp_style_is( 'my-style', 'registered' ) );
-		$this->assertEquals( 'my-script', Assets::init()->get( 'my-script' )->get_slug() );
-		$this->assertEquals( 'my-style', Assets::init()->get( 'my-style' )->get_slug() );
+		$this->existence_assertions( 'my' );
+	}
+
+	/**
+	 * @test
+	 */
+	public function it_should_locate_minified_versions_of_external_assets() {
+		if ( defined( SCRIPT_DEBUG ) ) {
+			$old_script_debug = SCRIPT_DEBUG;
+			uopz_undefine( 'SCRIPT_DEBUG' );
+		}
+
+		Asset::add( 'fake-script', 'fake.js' )->register();
+		Asset::add( 'fake-style', 'fake.css' )->register();
+		Asset::add( 'fake2-script', 'fake2.js' )->register();
+		Asset::add( 'fake2-style', 'fake2.css' )->register();
+		Asset::add( 'fake3-script', 'fake3.js' )->register();
+		Asset::add( 'fake3-style', 'fake3.css' )->register();
+
+		$slugs = [
+			'fake' => [ true, false ],
+			'fake2' => [ false, false ],
+			'fake3' => [ true, true ]
+		];
+
+		foreach ( array_keys( $slugs ) as $slug ) {
+			$this->existence_assertions( $slug );
+		}
+
+		foreach ( $slugs as $slug => $data ) {
+			$this->assert_minified_found( $slug, true, ...$data );
+			$this->assert_minified_found( $slug, false, ...$data );
+		}
+
+		if ( isset( $old_script_debug ) ) {
```

Comment:
This must be moved to an `@after` method.
If any of the assertions running before this line fail, then this code will never be executed and the redefinition will leak to following tests.